### PR TITLE
Add setPrompted API

### PR DESCRIPTION
**Pull request by anotherminh**
## Related Issues

- Part of https://transcend.height.app/T-18711

## Security Implications

_[none]_

## System Availability

_[none]_

### CODE REVIEW

File: src/core.ts [hunk 1]
```diff
@@ -356,6 +358,8 @@ export type TrackingConsentDetails = {
   confirmed: boolean;
   /** Consent last-modified timestamp (ISO 8601) */
   timestamp: string;
+  /** Whether or not the UI has been shown to the end-user (undefined in older versions of airgap.js) */
+  isPrompted: boolean | undefined;
```
Comment (eligrey) [inline src/core.ts:362]: the grammar here doesn't make sense. It reads as 'consent is prompted' (confusing). I'd much prefer a plain `prompted` field.

```suggestion
  prompted?: boolean;
```

Comment (anotherminh) [reply src/core.ts]: yeah good point, fixed!

**Review by eligrey — CHANGES_REQUESTED** [review]
This also needs `ConsentOptions.prompted`

**Review by eligrey — APPROVED** [review]

File: src/core.ts [hunk 2]
```diff
@@ -6,7 +6,6 @@ import { valuesOf } from '@transcend-io/type-utils';
 
 // local
 import { ConfigurablePurpose, SpecialTrackingPurpose } from './enums';
-import { boolean } from 'fp-ts';
```
Comment (anotherminh) [inline src/core.ts]: lol thanks @eligrey

**Pull request merged**